libs/BLEKit/include/CoreGap.h

```diff
@@ -24,11 +33,12 @@ class CoreGap
 	void onInitializationComplete(BLE::InitializationCompleteCallbackContext *params);
 	// void onInit(std::function<void()> cb) { _post_init = cb; }
 
-	void setDeviceName(const char *name);
-
 	void startAdvertising();
+	void refreshAdvertising(CustomAdvertising custom_advertising);
 
   private:
+	void setAdvertisingData(CustomAdvertising data);
+
```

Comment:
`refresh` is a bit strange. it gives the false impression that it's going to pull the new data on it's own, like when you refresh a page in your browser.

I would have only one method, `setAdvertisingData`. 

if they change, we just set them again.

Reply:
`refresh` refers to refresh dynamically the data in advertising.

The `setAdvertisingData` used in CoreGap will set in local only the data from the struct into `advertising_data_builder` from mbed

`refresh` will do this plus a call to `_gap.setAdvertisingPayload(_advertising_handle, _advertising_data_builder.getAdvertisingData());` to set data from local to "online"

Reply:
Maybe I'm missing something, but I don't see a case where you want to refresh the offline data but not make them available to the outside world.

I would refactor CoreGap to have setAdvertisingData do the offline + online work, if that makes sens.

This can be tackled in an upcoming PR of course

Reply:
If you just want to replace the name `refresh` by `setAdvertisingData`, no problem